### ISSUE TITLE
Installation error

### ISSUE THREAD

**Issue by Denoder** [adrenth/rssfetcher#10]
Im getting this issue when installing:
![](http://image.prntscr.com/image/8f2cf85e8ba1498dbb26f480144928b3.png)

**Comment by adrenth** [comment]
I've pushed an update. Cannot reproduce, need some more details.

Can you tell me:
- which MySQL version you are using?
- if you have tried `php artisan plugin:refresh Adrenth.RssFetcher` (CAUTION: this will probably delete some plugin data).
- Is the MySQL InnoDB driver installed/present?

~ Alwin

**Comment by Denoder** [comment]
Maria DB 10.0
Yes ive tried and it still gives the same error
yes InnoDB is installed
![](http://image.prntscr.com/image/4591d6e078984a2dbbb4072be3d6c338.png)

**Comment by adrenth** [comment]
Hey, thanks for pasting those exceptions, that makes a lot of sense to me. This should be an easy fix :-) My database configuration allows me to have longer keynames.  I have modified the length of the keys, so this issue should be resolved.

Make sure you have installed version 1.0.14.

And please if you like this plugin, give me some feedback by rating this plugin on octobercms.com. Thanks.

**Comment by Denoder** [comment]
issue still occrs for me :(

**Comment by adrenth** [comment]
The build of version 1.0.14 was not triggered by OctoberCMS' Marketplace. It should be released by now. Please install 1.0.14 which addresses this issue.

**Comment by Denoder** [comment]
im still getting:
```
SQLSTATE[HY000]: General error: 1709 Index column size too large. The maximum column size is 767 bytes. (SQL: alter table `adrenth_rssfetcher_items` add unique `item_id_unique`(`item_id`))
```

**Comment by adrenth** [comment]
The index which the migration file is trying to create is named `item_id_unique` for the column `item_id`. 

This is basic SQL and this is correct: 
````
alter table `adrenth_rssfetcher_items` add unique `item_id_unique`(`item_id`))
````

This is de PHP code from the migration file:
````
$table->string('item_id')->unique('item_id_unique'); // VARCHAR(255)
````

The cause of this issue is that you are using a different collation type on your database than the common `utf8`. I think you have created the database with a `utf8mb4` (4-bytes character size). With `utf8` the database uses 3 bytes per character (255*3 = 765 bytes). With `utf8mb4` collation each character takes up to 4 bytes (255*4 = 1020) which exceeds the setting of 767 bytes.

I have changed the length of the fields to 191 characters (4*191=764 bytes). Which should be fine.

This issue is addressed in release 1.0.15. Please let me know if this works for you.

Please rate my plugin if you like it on octobercms.com. Thanks!

**Comment by Denoder** [comment]
still getting:
```
"SQLSTATE[HY000]: General error: 1005 Can't create table `sata_database`.`#sql-ea6_331a` (errno: 150 "Foreign key constraint is incorrectly formed") (SQL: alter table `adrenth_rssfetcher_items` add constraint items_source_id_foreign foreign key (`source_id`) references `adrenth_rssfetcher_sources` (`id`) on delete cascade)" on line 666 of 
```
